[docs] improve how sui works doc (#8492)
MystenLabs/sui: doc/src/learn/how-sui-works.md

Finally, we note that the role of quorum driver does not require access to any private keys and can safely be delegated to a third party (e.g., a full node, an RPC provider, a custodial wallet).

## Transactions on shared objects

Many use-cases require *shared* objects that can be touched by two or more addresses at once--e.g., an auction with open bidding or a central limit order book accepting arbitrary trades. In such cases, Sui must sequence transactions touching the same shared object(s) using a [consensus](architecture/consensus.md) protocol. Sui uses [Narwhal](https://arxiv.org/abs/2105.11827) for high-throughput, horizontally scalable transaction dissemination and [Bullshark](https://arxiv.org/abs/2209.05633) for zero message overhead consensus.

For transactions involving one or more shared objects, the process is as described above up to step (4), where instead:
1. Upon receiving a cerficate, the validator (afer checking its validity) uses Narwhal to submit the certificate to Bullshark for sequencing.

> 1. Once the transaction has been sequenced, it is executed to produce effects using same flow as (5) above.

Reviewer: ```suggestion
1. After Sui sequences the transaction, it executes it to produce to produce effects using same flow as (5) above.
```

Reviewer 2: there is a dupllication of "to produce"

> 1. Once the transaction has been sequenced, it is executed to produce effects using same flow as (5) above.

Reviewer 2: ```suggestion
2. After Sui sequences the transaction, it executes it to produce effects using same flow as (5) above.
```
